modules/custom/activity_logger/src/Service/ActivityLoggerFactory.php

```diff
@@ -175,4 +176,43 @@ protected function createFieldInstances($message_type, array $fields) {
     }
   }
 
+  /**
+   * Checks if a message already exists.
+   *
+   * @param string $message_type
+   *   The message type.
+   * @param string $context
+   *   The context of the message.
+   * @param array $destination
+   *   The array of destinations to check for, include delta as well.
+   * @param array $related_object
+   *   The related object, include target_type and target_id in array.
+   * @param string $uid
+   *   The uid of the message.
+   *
+   * @return int
+   *   Returns true if the message exists.
+   */
+  public function checkIfMessageExist($message_type, $context, array $destination, array $related_object, $uid) {
+    $exists = FALSE;
+
+    $query = \Drupal::entityQuery('message');
```

Comment:
@jaapjan we should use dependency injection here.

Reply:
Agree, but we don't use it for the rest of the services in here as well. So I propose to work on that in a later stage (since this is considered "hotfix")